connectors/es/management_client.py

```diff
@@ -185,11 +186,40 @@ async def clean_index(self, index_name):
             )
         )
 
-    async def list_indices(self, index=None):
-        return await self._retrier.execute_with_retry(
+    async def list_indices(self, index="*"):
+        """
+        List indices using Elasticsearch.stats API. Includes the number of documents in each index.
+        """
+        indices = {}
+        response = await self._retrier.execute_with_retry(
             partial(self.client.indices.stats, index=index)
         )
 
+        for index in response["indices"].items():
+            indices[index[0]] = {"docs_count": index[1]["primaries"]["docs"]["count"]}
+
+        return indices
+
+    async def list_indices_serverless(self, index="*"):
+        """
+        List indices in a serverless environment. This method is a workaround to the fact that
+        the `indices.stats` API is not available in serverless environments.
+        """
+
+        indices = {}
+        try:
+            response = await self._retrier.execute_with_retry(
+                partial(self.client.indices.get, index=index)
+            )
```

Comment:
So there's absolutely no way to do so? What if we do N+1 query here, doing count on each index instead?

Reply:
I was thinking about the N+1 approach here but prefer not to because of the performance issues. I assume that getting indices names quickly is better than getting the names and docs counts slowly.